node/src/main/scala/org/bitcoins/node/db/NodeDbManagement.scala

```diff
@@ -1,12 +1,23 @@
 package org.bitcoins.node.db
 
-import org.bitcoins.db.DbManagement
-import org.bitcoins.node.models.BroadcastAbleTransactionTable
-import slick.lifted.TableQuery
+import org.bitcoins.db.{DbManagement, JdbcProfileComponent}
+import org.bitcoins.node.config.NodeAppConfig
+import org.bitcoins.node.models.{BroadcastAbleTransactionDAO}
 
-object NodeDbManagement extends DbManagement {
+import scala.concurrent.ExecutionContext
 
-  private val txTable = TableQuery[BroadcastAbleTransactionTable]
+trait NodeDbManagement extends DbManagement { _: JdbcProfileComponent =>
+
+  import profile.api._
+
+  def ec: ExecutionContext
+
+  override def appConfig: NodeAppConfig
+
+  private lazy val txTable: TableQuery[Table[_]] = {
+    BroadcastAbleTransactionDAO()(appConfig.asInstanceOf[NodeAppConfig], ec).table
```

Comment:
`appConfig` is already an instance of `NodeAppConfig`

Reply:
Yes.... this is sooo confusing to me. I don't understand why this won't compile. For reference, if I don't have these casts in all of my `*DbManagement.scala` files I get: 

>[error] /home/suredbits/dev/bitcoin-s/node/src/main/scala/org/bitcoins/node/db/NodeDbManagement.scala:18:35: type mismatch;
[error]  found   : org.bitcoins.db.AppConfig
[error]  required: org.bitcoins.node.config.NodeAppConfig
[error]     BroadcastAbleTransactionDAO()(appConfig, ec).table
[error]                                   ^
[error] one error found